src/components/annotations/draw.js

```diff
@@ -524,21 +532,17 @@ function drawOne(gd, index) {
 
                         update[annbase + '.x'] = xa ?
                             xa.p2r(xa.r2p(options.x) + dx) :
-                            ((headX + dx - gs.l) / gs.w);
+                            (options.x + (dx / gs.w));
```

Comment:
motivated by the fact that the old way would have needed to explicitly account for shift - notice though in the [`showarrow: false` case](https://github.com/plotly/plotly.js/pull/1590/files#diff-05274991eb85aefd482f72514dbda1a5R601) I did need to just include shift, due to the auto-anchor business...